plugins/header_rewrite/operators.cc

```diff
@@ -692,10 +692,15 @@ OperatorRMCookie::exec(const Resources &res) const
     int cookies_len     = 0;
     const char *cookies = TSMimeHdrFieldValueStringGet(res.bufp, res.hdr_loc, field_loc, -1, &cookies_len);
     std::string updated_cookie;
-    if (CookieHelper::cookieModifyHelper(cookies, cookies_len, updated_cookie, CookieHelper::COOKIE_OP_DEL, _cookie) &&
-        TS_SUCCESS ==
-          TSMimeHdrFieldValueStringSet(res.bufp, res.hdr_loc, field_loc, -1, updated_cookie.c_str(), updated_cookie.size())) {
-      TSDebug(PLUGIN_NAME, "OperatorRMCookie::exec, updated_cookie = [%s]", updated_cookie.c_str());
+    if (CookieHelper::cookieModifyHelper(cookies, cookies_len, updated_cookie, CookieHelper::COOKIE_OP_DEL, _cookie)) {
+      if (0 == strlen(updated_cookie.c_str())) {
```

Comment:
Perhaps `if (updated_cookie.empty())` would be better?

Reply:
+1

Reply:
Updated to use string.empty instead of strlen.